checker/src/org/checkerframework/checker/index/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -406,4 +406,10 @@ private AnnotationMirror checkForMathRandomSpecialCase(Tree randTree, Tree arrLe
         }
         return null;
     }
+
+    /** Checks if the expression is non-negative, i.e. it has Positive on NonNegative annotation. */
```

Comment:
You should use this method in the modulo PR too.

Reply:
And, should this replace `isNonNegative` in `LowerBoundTransfer`? At the very least, please leave a TODO there that we can come back to later.

Reply:
Not exactly - the parameter type is different; perhaps they could two overloads in one class.
I added the TODO.